testgroup.go

```diff
@@ -235,12 +235,14 @@ func findTestMethods(t *testing.T, group interface{}) []testMethod {
 }
 
 func requireGroupAndGroupPtrMethodsToMatch(t *testing.T, groupType reflect.Type) {
+	t.Helper()
+
 	if groupType.Kind() == reflect.Ptr {
 		return
 	}
 
 	ptrType := reflect.PtrTo(groupType)
-	if ptrType != nil && ptrType.NumMethod() != groupType.NumMethod() {
+	if ptrType.NumMethod() != groupType.NumMethod() {
```

Comment:
Is this guaranteed to be non-nil now when it wasn't before?

Reply:
I think it was always guaranteed to be non-`nil`, actually (see https://pkg.go.dev/reflect#PtrTo), and staticcheck caught that:

```
testgroup.go:245:5: SA4023: this comparison is always true (staticcheck)
	if ptrType != nil && ptrType.NumMethod() != groupType.NumMethod() {
	   ^
testgroup.go:244:13: SA4023(related information): the lhs of the comparison is the 1st return value of this function call (staticcheck)
	ptrType := reflect.PtrTo(groupType)
	           ^
..../reflect/type.go:1444:6: SA4023(related information): reflect.PtrTo never returns a nil interface value (staticcheck)
func PtrTo(t Type) Type { return PointerTo(t) }
     ^
```